presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/TransformQuantifiedComparisonApplyToScalarApply.java

```diff
@@ -191,10 +221,32 @@ private PlanNode rewriteQuantifiedOrderableApplyNode(ApplyNode node, QuantifiedC
                     node.getId(),
                     context.rewrite(node.getInput()),
                     subqueryPlan,
-                    Assignments.of(subValue, subValue.toSymbolReference()),
+                    Assignments.of(extremeValue, extremeValue.toSymbolReference()),
                     node.getCorrelation());
 
-            ComparisonExpression valueComparedToSubquery = new ComparisonExpression(quantifiedComparison.getComparisonType(), quantifiedComparison.getValue(), subValue.toSymbolReference());
+            BooleanLiteral emptySetResult = quantifiedComparison.getQuantifier().equals(ALL) ? TRUE_LITERAL : FALSE_LITERAL;
```

Comment:
put the whole section which build the expression into separate method with a comment what expression it builds